Library/Homebrew/formula.rb

```diff
@@ -1543,7 +1543,11 @@ def std_configure_args
   # Standard parameters for cargo builds.
   sig { params(root: T.any(String, Pathname), path: String).returns(T::Array[T.any(String, Pathname)]) }
   def std_cargo_args(root: prefix, path: ".")
-    ["--locked", "--root", root, "--path", path]
+    ["--locked", "--root", root, "--path", path,
+     "--config", "target.x86_64-apple-darwin.rustflags=['-Ctarget-cpu=nehalem']",
+     "--config", "target.aarch64-apple-darwin.rustflags=['-Ctarget-cpu=apple-m1']",
```

Comment:
Use `Hardware::CPU.optimization_flags.fetch(Hardware.oldest_cpu)` to get the right `nehalem` here. May want new `rust_optimization_flags` method in `hardware.rb` that handles the `apple-m1` case.

Reply:
Does `Hardware::CPU.optimization_flags.fetch(Hardware.oldest_cpu)` always return intel CPU?

Reply:
Sample output that may help:

```
==> Interactive Homebrew Shell
Example commands available with: `brew irb --examples`

WARNING: This version of ruby is included in macOS for compatibility with legacy software. 
In future versions of macOS the ruby runtime will not be available by 
default, and may require you to install an additional package.

brew(main):001:0> Hardware::CPU.optimization_flags.fetch(Hardware.oldest_cpu)
=> ""
brew(main):002:0> Hardware::CPU.optimization_flags
=> {:native=>"-march=native", :ivybridge=>"-march=ivybridge", :sandybridge=>"-march=sandybridge", :nehalem=>"-march=nehalem", :core2=>"-march=core2", :core=>"-march=prescott", :arm_vortex_tempest=>"", :armv6=>"-march=armv6", :armv8=>"-march=armv8-a", :ppc64=>"-mcpu=powerpc64", :ppc64le=>"-mcpu=powerpc64le"}
brew(main):003:0> Hardware.oldest_cpu
=> :arm_vortex_tempest
```

Reply:
What you are suggesting moves cpu-selesction out of cargo and into Homebrew, because I'm forced to use  `build.rustflags` instead of `target.<triple>.rustflags`, since as far as I can see `Hardware.oldest_cpu` can only be called with host arch and macos version.
This shouldn't be a problem. The following map seems to suffice (trigger warning: pseudocode):
```Ruby
:native => 'native',
:nehalem => 'nehalem',
:arm_vortex_tempest => 'apple-m1',
_ => 'generic'
# or
_ => _.to_string()
```
`:arm_vortex_tempest => 'apple-m1` Does this seem correct?

Reply:
I think could still use `target.*.rustflags` but just want to make sure the values there are stored in Ruby rather than hardcoded strings.